src/python/pants/option/option_tracker.py

```diff
@@ -8,22 +8,23 @@
 from pants.option.ranked_value import Rank
 
 
+@dataclass(frozen=True)
+class OptionHistoryRecord:
+    value: str
+    rank: Rank
+    deprecation_version: Optional[str]
+    details: Optional[str]
+
+
 # TODO: Get rid of this? The parser should be able to lazily track.
 class OptionTracker:
```

Comment:
Is this v1 only?

Reply:
I've never 100% understood this class, because I thought that the `RankedValue` container would be able to record where things had come from. But have never looked deeply.

Reply:
The `RankedValue` is just a single value attached to a Rank, it doesn't know about any other values.

Reply:
But there's probably some simplification we can do. E.g., the RankedValue can carry the "details" information (e.g., which config file a CONFIG-level value came from)

Reply:
It is, but it is inside an `OptionValueContainer`, which does have all of them, I thought? Maybe not.